misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -35,16 +36,14 @@ sub run {
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $verbose      = $ref_arg->{verbose} || 0;
+  my $dbi          = $ref_arg->{dbi} || $self->dbi;
 
   if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
     croak "Need to pass source_id, species_id and files as pairs";
   }
-  $verbose |=0;
 
-  my $file = @{$files}[0];
+  my $file = shift @{$files};
 
   my $mgi_io = $self->get_filehandle($file);
 
```

Comment:
GitHub is not letting me attach a comment to line 51 because it is too far from any changes so I'll mark it here: please croak instead of printing to STDERR and returning 1.